Fixed preview bug with circle counter and added ccache

borgmanJeremy opened the pull request:
Closes #942

hosiet commented:
Looks like the thickness was doubled when compared with the original implementation (`m_thickness * 2`). Is that the desired behaviour?

Besides, I would suggest splitting the ccache part into a separate PR/commit.

borgmanJeremy commented:
Sorry I was tired when I committed this :) I split the changes to circle counter into the other PR and am merging the cmake stuff separately. I added some comments in the code to explain the *2 (it is related to radius vs diameter).
